docs/source/channels.rst

```diff
@@ -0,0 +1,253 @@
+in that independent worker clients can build up a set of results
+which can be read later by a different client.
+This opens up Dask/Distributed to being integrated in a wider application
+environment similar to other python task queues such as Celery_.
+
+.. _local client: http://distributed.readthedocs.io/en/latest/task-launch.html#submit-tasks-from-worker
+.. _Celery: http://www.celeryproject.org/
+
+.. code-block:: python
+
+    import random, time, operator
+    from distributed import Client, local_client
+    from time import sleep
+
+    def emit(name):
+        with local_client() as c:
+           chan = c.channel(name)
+           while True:
+               future = c.submit(random.random, pure=False)
+               chan.append(future)
+               sleep(1)
+
+    def combine():
+        with local_client() as c:
+            a_chan = c.channel('a')
+            b_chan = c.channel('b')
+            out_chan = c.channel('adds')
+            for a, b in zip(a_chan, b_chan):
+                future = c.submit(operator.add, a, b)
+                out_chan.append(future)
+
+    client = Client()
+
+    emitters = (client.submit(emit, 'a'), client.submit(emit, 'b'))
+    combiner = client.submit(combine)
+    chan = client.channel('adds')
+
+
+    for future in chan:
+        print(future.result())
+       ...:
+    1.782009416831722
+    ...
+
+All iterations on a channel by different clients can be stopped using the ``stop`` method
+
+.. code-block:: python
+
+    chan.stop()
+
+
+Very short-lived clients
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you wish to submit work to your cluster from a short lived client such as a
+web application view, an AWS Lambda function or some other fire and forget script,
+channels give a way to do this.
+
+Motivating Example - A workflow designer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine the following deployment scenario:
+
+.. image:: images/channels_example.png
+    :alt: Workflow designer application
+
+1) A `Custom Dask Graph`_ is designed in a web app, perhaps inserting code snippets.
+
+.. code-block:: javascript
+
+    {
+        "tracking_key_1" :["allowed_module.func", arg1, arg2],
+        "tracking_key_2" :["allowed_module.func", "tracking_key_1"]
+    }
+
+2) Import strings or code snippets are validated and imported to function objects.
+The graph might be persisted and appropriate keys added for tracking. The graph is
+added to a channel that is followed by one of the clients. As the client will
+only exist for the lifetime of the request then the graph is transmitted using scatter.
+
+.. code-block:: python
+
+    client = Client('scheduler-address:8786')
+    chan = client.channel('input')
+    dask_graph = {
+                    "tracking_key_1" :(func, arg1, arg2),
+                    "tracking_key_2" :(func2, "tracking_key_1")
+                 }
+    future_list = client.scatter([("<worflow_run_id>", dask_graph)])
+    chan.append(future_list[0])
+    chan.flush()
```

Comment:
Would it be reasonable to just submit the task now rather than send it to someone else to submit?